File: backend/internal/domain/consent.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// Consent is an LGPD consent record scoped to a campus. Revoked rows are
// preserved as the registry trail (docs/13); re-granting creates a new row.
type Consent struct {
	ID              uuid.UUID  `json:"id"`
	PersonID        uuid.UUID  `json:"person_id"`
	ConsentType     string     `json:"consent_type"`
	ConsentVersion  string     `json:"consent_version"`
	Purpose         string     `json:"purpose"`
	GrantedAt       time.Time  `json:"granted_at"`
	GrantedByPerson *uuid.UUID `json:"granted_by_person_id"`
	SignatureData   *string    `json:"signature_data"`
	IsActive        bool       `json:"is_active"`
	RevokedAt       *time.Time `json:"revoked_at"`
	RevokedReason   *string    `json:"revoked_reason"`
	CampusID        uuid.UUID  `json:"campus_id"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
}

// ConsentTypes enumerates the valid LGPD consent categories.
var ConsentTypes = []string{"DATA_PROCESSING", "IMAGE_USAGE", "HEALTH_DATA", "MINOR_GUARDIAN"}

File: frontend/src/types/consent.ts
export type ConsentType =
  | 'DATA_PROCESSING'
  | 'IMAGE_USAGE'
  | 'HEALTH_DATA'
  | 'MINOR_GUARDIAN';

export interface Consent {
  id: string;
  person_id: string;
  consent_type: ConsentType;
  consent_version: string;
  purpose: string;
  granted_at: string;
  granted_by_person_id: string | null;
  signature_data: string | null;
  is_active: boolean;
  revoked_at: string | null;
  revoked_reason: string | null;
}

export interface CreateConsentInput {
  person_id: string;
  consent_type: ConsentType;
  purpose: string;
  consent_version?: string;
  granted_by_person_id?: string;
  signature_data?: string;
}

export interface ConsentListResponse {
  data: Consent[];
}
